pkgs/misc/uboot/default.nix

```diff
@@ -4,24 +4,27 @@
 }:
 
 let
-  buildUBoot = { version ? "2019.10"
-            , filesToInstall
-            , installDir ? "$out"
-            , defconfig
-            , extraConfig ? ""
-            , extraPatches ? []
-            , extraMakeFlags ? []
-            , extraMeta ? {}
-            , ... } @ args:
-           stdenv.mkDerivation ({
-
+  defaultVersion = "2019.10";
+  defaultSrc = fetchurl {
+    url = "ftp://ftp.denx.de/pub/u-boot/u-boot-${defaultVersion}.tar.bz2";
+    sha256 = "053hcrwwlacqh2niisn0zas95zkbffw5aw5sdhixs8lmfdq60vcd";
+  };
+  buildUBoot = {
+    version ? null
+  , src ? null
+  , filesToInstall
+  , installDir ? "$out"
+  , defconfig
+  , extraConfig ? ""
+  , extraPatches ? []
+  , extraMakeFlags ? []
+  , extraMeta ? {}
+  , ... } @ args: stdenv.mkDerivation ({
     pname = "uboot-${defconfig}";
     inherit version;
```

Comment:
```
error: attribute 'version' at /Users/samuel/tmp/nixpkgs/nixpkgs-PR74363/pkgs/misc/uboot/default.nix:26:5 already defined at /Users/samuel/tmp/nixpkgs/nixpkgs-PR74363/pk
gs/misc/uboot/default.nix:24:12
```

Reply:
right, missed that during the refactoring. Updated.